std/socket.d

```diff
@@ -1663,6 +1663,23 @@ public:
     }
 
     /**
+     * Compares with another InternetAddress of same type for equality
+     * Returns: true if the InternetAddresses share the same address and
+     * port number.
+     * Examples:
+     * --------------
+     * InternetAddress addr1,addr2;
+     * if (addr1 == addr2) { }
+     * --------------
+     */
+    override bool opEquals(Object o) const
+    {
+        auto other = cast(InternetAddress)o;
+        return other && this.sin.sin_addr.s_addr == other.sin.sin_addr.s_addr &&
```

Comment:
That's a good thing to discuss.  It depends on how opEquals is supposed to be used. If it's valid to pass other non-InternetAddress objects then it is necessary.  If not, then we could use a contract to validate that o is of type InternetAddress. However,  I plan on submitting an enhancement request to change the '==' operator to use a typed opEquals instead of the generic one, which would mean we wouldn't need to cast the Object parameter cause we could just use an InternetAddress directly.